src/api/v1beta1/properties.go

```diff
@@ -347,14 +351,21 @@ func (dk *DynaKube) CodeModulesVersion() string {
 	if !dk.CloudNativeFullstackMode() && !dk.ApplicationMonitoringMode() {
 		return ""
 	}
-	if dk.CodeModulesImage() != "" {
-		codeModulesImage := dk.CodeModulesImage()
+	if dk.CustomCodeModulesImage() != "" {
+		codeModulesImage := dk.CustomCodeModulesImage()
 		return getRawImageTag(codeModulesImage)
 	}
-	if dk.Version() != "" && !dk.CloudNativeFullstackMode() {
-		return dk.Version()
+	if dk.CustomCodeModulesVersion() != "" {
```

Comment:
we have to keep the check `!dk.CloudNativeFullstackMode()` there will be another field for codeModulesVersion in cloudNative